compiler/rustc_middle/src/mir/coverage.rs

```diff
@@ -158,18 +158,36 @@ pub struct Expression {
     pub rhs: CovTerm,
 }
 
+#[derive(Clone, Debug)]
+#[derive(TyEncodable, TyDecodable, Hash, HashStable, TypeFoldable, TypeVisitable)]
+pub enum MappingKind {
+    /// Associates a normal region of code with a counter/expression/zero.
+    Code(CovTerm),
+}
+
+impl MappingKind {
+    /// Iterator over all coverage terms in this mapping kind.
+    pub fn terms(&self) -> impl Iterator<Item = CovTerm> {
+        let one = |a| std::iter::once(a);
+        match *self {
+            Self::Code(term) => one(term),
+        }
```

Comment:
I don't think this will work with variants that have multiple terms unless all of them are slices and you use https://doc.rust-lang.org/std/slice/fn.from_ref.html here (in which case you could proabably return a slice instead of an iterator)

But also not important, you can refactor to make it work when you add more variants

Reply:
All of the variants supported by LLVM have exactly zero, one, or two terms.

So my plan is to eventually change this definition to something like:
```rust
let one = |a| [Some(a), None].into_iter().flatten()
```

That way the individual arms that use `one` won't have to change.

(But I won't bother doing this until I actually add a variant that requires 0 or 2 terms.)